File: VIPER Tests.xctemplate/Objective-C/___FILEBASENAME___ViewTests.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

#import <UIKit/UIKit.h>
#import <XCTest/XCTest.h>

#import "___FILEBASENAMEASIDENTIFIER___View.h"

#import "___FILEBASENAMEASIDENTIFIER___Wireframe.h"
#import "___FILEBASENAMEASIDENTIFIER___ViewProtocols.h"

@interface ___FILEBASENAMEASIDENTIFIER___ViewTests : XCTestCase <___FILEBASENAMEASIDENTIFIER___PresenterInterface>

@property (nonatomic, strong) ___FILEBASENAMEASIDENTIFIER___View *view;
@property (nonatomic, strong) UIWindow *window;

// Test Objects
@property (nonatomic, strong) XCTestExpectation *expectation;

@end

@implementation ___FILEBASENAMEASIDENTIFIER___ViewTests

- (void)setUp {
        [super setUp];
	
        UIStoryboard *sb =[UIStoryboard storyboardWithName:k___FILEBASENAMEASIDENTIFIER___Storyboard bundle:[NSBundle mainBundle]];
        self.view = [sb instantiateViewControllerWithIdentifier:k___FILEBASENAMEASIDENTIFIER___ViewIdentifier];
	[self.view loadView];
	self.window = [UIWindow new];
        self.window.rootViewController = self.view;
        [self.window makeKeyAndVisible];

        self.view.presenter = self;
}

- (void)tearDown {
        [super tearDown];
        self.expectation = nil;
}

#pragma mark - Operational

#pragma mark - View Interface

#pragma mark - Presenter Interface

@end

File: VIPER Tests.xctemplate/Swift/___FILEBASENAME___ViewTests.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

import UIKit
import XCTest

@testable import ___PROJECTNAME___

class ___FILEBASENAMEASIDENTIFIER___ViewTests: XCTestCase {
        var view = ___FILEBASENAMEASIDENTIFIER___View()

        // MARK: - Test Objects
        var expectation: XCTestExpectation?

        override func setUp() {
                super.setUp()

                _ = view.view

                view.presenter = self
        }

        override func tearDown() {
                super.tearDown()
                expectation = nil
        }

        // MARK: - Operational

        // MARK: - Presenter to View Interface

}

// MARK: - View To Presenter Interface
extension ___FILEBASENAMEASIDENTIFIER___ViewTests: ___FILEBASENAMEASIDENTIFIER___ViewToPresenterInterface {

}
